bigtop_toolchain/manifests/ruby.pp

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class bigtop_toolchain::ruby {
+
+  require bigtop_toolchain::packages
+
+  $url = "https://cache.ruby-lang.org/pub/ruby/2.4"
+
+  $ruby24 = "ruby-2.4.0.tar.gz"
+  $ruby24dir = "ruby-2.4.0"
+
+  exec { "download ruby":
+     cwd  => "/usr/src",
+     command => "/usr/bin/wget $url/$ruby24 && mkdir -p $ruby24dir && /bin/tar -xvzf $ruby24 -C $ruby24dir --strip-components=1 && cd $ruby24dir",
+     creates => "/usr/src/$ruby24dir",
+  }
+
+  exec { "install ruby":
+     cwd => "/usr/src/$ruby24dir",
+     command => "/usr/src/$ruby24dir/configure --prefix=/usr/local --disable-shared && /usr/bin/make install",
+     creates => "/usr/local/bin/ruby",
+     timeout => 1800,
+     require => EXEC["download ruby"]
+  }
```

Comment:
So ruby 2.4 version is the only one that works with log stash? Or we can install it from Distro's pkg manager (yum/apt)?

Reply:
Yea, for the dependency issues, the ruby-2.4 is the latest version that that works with logstash-5.4.1.
And the bigtop should support different kind of distros and platforms(x86/arm64). 
It may not be guaranteed the various distros could offer the ruby-2.4 pkg (x86/arm64) installation. IMHO, it's robust to build the ruby-2.4 binaries from source though it may cost extra time in bigtop_toolchain build.

Reply:
Sounds good to me. I'm a +1 but not an expert on Logstash. If no one chimed in I'll commit this in this week. 
Ping me if not. Thanks!